packages/remix-server-runtime/server.ts

```diff
@@ -413,22 +305,15 @@ async function handleDocumentRequest(
     context.errors = sanitizeErrors(context.errors, serverMode);
   }
 
-  let statusCode: number;
-  let headers: Headers;
-  if (build.future.unstable_singleFetch) {
-    let merged = mergeResponseStubs(context, responseStubs);
-    statusCode = merged.statusCode;
-    headers = merged.headers;
+  let merged = mergeResponseStubs(context, responseStubs);
+  let statusCode = merged.statusCode;
+  let headers = merged.headers;
 
-    if (isRedirectStatusCode(statusCode) && headers.has("Location")) {
-      return new Response(null, {
-        status: statusCode,
-        headers,
-      });
-    }
-  } else {
-    statusCode = context.statusCode;
-    headers = getDocumentHeaders(build, context);
```

Comment:
I also removed the now unused `headers.ts` file and the set-cookie-parser dependency in [531b286](https://github.com/remix-run/react-router/pull/11522/commits/531b28645c2736c2db70e2b58e5d4e67e7c9f727)